### PR TITLE
✨Source Zendesk Support: New Stream - Organization Fields

### LINKED ISSUE #16871: Source Zendesk Support: new stream Organization Fields
## Tell us about the problem you're trying to solve
You can use this API to add fields to the Organization page in the Zendesk user interface. Basic text fields, date fields, as well as customizable drop-down and number fields are available. The fields correspond to the [Organization Fields](https://developer.zendesk.com/api-reference/ticketing/organizations/organization_fields/) that admins can add using the Zendesk admin interface. 

**The fields are only visible to agents and admins.**

[Allowed for:](https://developer.zendesk.com/api-reference/ticketing/organizations/organization_fields/#allowed-for)
- Agents

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes #16871

## Enhancement
New streams: 

- Organization Fields

## Recommended reading order
1. `source_zendesk_support/source.py`
2. `source_zendesk_support/streams.py`
3.  `source_zendesk_support/schemas/*`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter [Yes]

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation which references the generator is updated as needed
+

### CODE REVIEW

**Comment by btkcodedev** [comment]
Update: Awaiting creds for adding expected records

File: airbyte-integrations/connectors/source-zendesk-support/unit_tests/unit_test.py
Comment (archangelic) [inline airbyte-integrations/connectors/source-zendesk-support/unit_tests/unit_test.py:434]: Missing comma
```suggestion
            (OrganizationFields, {}, {"updated_at": "2022-03-17T16:03:07Z"}, {"updated_at": "2022-03-17T16:03:07Z"}),
```

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/organization_fields.json
Comment (archangelic) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/organization_fields.json:6]: Please add the following missing properties: `custom_field_options, relationship_filter, relationship_target_type, system, tag` per [the documentation](https://developer.zendesk.com/api-reference/ticketing/organizations/organization_fields/#json-format).

Also, the indentation needs to be fixed.

**Review by archangelic — CHANGES_REQUESTED** [review]
Thanks for your contribution. I have listed out some recommended changes.

**Comment by marcosmarxm** [comment]
@btkcodedev are you planning to continue this contribution?

**Comment by archangelic** [comment]
Pushed more fixes. Still have some unit tests that are blocking completion.

**Review by darynaishchenko — APPROVED** [review]

**Review by darynaishchenko — COMMENTED** [review]
please, merge conflicts and if CI checks are still running you can close and reopen PR to fix it.

**Review by darynaishchenko — CHANGES_REQUESTED** [review]
could you add expected_records for this stream?

**Comment by btkcodedev** [comment]
Merge conflicts cleared, 
Tried to generate data to the endpoint via cURL, but has scope issue:
Error ref: `{
    "error": "Forbidden",
    "description": "You are missing the following required scopes: write"
}`

File: airbyte-integrations/connectors/source-zendesk-support/unit_tests/unit_test.py
Comment (darynaishchenko) [inline airbyte-integrations/connectors/source-zendesk-support/unit_tests/unit_test.py:153]: Just for my understanding, why with identical  ticket_forms_response and status_code we receive different number of streams? 20 and 27 for 200 status code, 19 and 24 for 403 status code

**Comment by btkcodedev** [comment]
@darynaishchenko Unit test fixed, Please continue further review :)

**Comment by darynaishchenko** [comment]
@btkcodedev Hi, sorry for delay, could you please merge conflicts.

**Comment by btkcodedev** [comment]
@darynaishchenko Conflicts cleared :)

**Comment by bmcgilli** [comment]
@darynaishchenko can you please follow up? @lazebnyi

**Review by darynaishchenko — APPROVED** [review]

**Comment by darynaishchenko** [comment]
@archangelic Hi, could you please review this PR?  it's blocked by your approve.

**Review by archangelic — APPROVED** [review]

**Comment by archangelic** [comment]
@darynaishchenko Sorry about that. I've approved!

**Comment by btkcodedev** [comment]
@darynaishchenko Please continue the review as it got two approvals :)

**Pull request closed without merging**
